src/models/logic/transaction.ts

```diff
@@ -386,65 +391,129 @@ export async function removeOutdatedPendings(
     }
 }
 
-async function getTransactionHashes(params: {
-    address?: string | null;
-    assetType?: H160 | null;
+async function getHashesByPlatformAddress(params: {
+    address: string;
     page: number;
     itemsPerPage: number;
-    type?: string[] | null;
-    includePending?: boolean | null;
 }): Promise<string[]> {
-    const where: Sequelize.WhereOptions<TransactionAttribute> = {};
-    if (params.type != null) {
-        where.type = { [Sequelize.Op.in]: params.type };
+    const { address, page, itemsPerPage } = params;
+    try {
+        return models.AddressLog.findAll({
+            attributes: ["transactionHash"],
+            where: {
+                address
+            },
+            order: [["blockNumber", "DESC"], ["transactionIndex", "DESC"]],
+            limit: itemsPerPage,
+            offset: (page - 1) * itemsPerPage
+        }).map(r => r.get("transactionHash"));
+    } catch (err) {
+        console.error(err);
+        throw Exception.DBError();
     }
-    if (params.includePending !== true) {
-        where.isPending = false;
+}
+
+async function getHashesByAssetAddress(params: {
+    address: string;
+    assetType?: string | null;
+    page: number;
+    itemsPerPage: number;
+}): Promise<string[]> {
+    const { address, assetType, page, itemsPerPage } = params;
+    try {
+        return models.AssetAddressLog.findAll({
+            attributes: ["transactionHash"],
+            where: {
+                address,
+                ...(assetType && { assetType })
+            },
+            order: [["blockNumber", "DESC"], ["transactionIndex", "DESC"]],
```

Comment:
Do we have indexes like ["address", "blockNumber", "transactionIndex"] and ["address", "assetType", "blockNumber", "transactionIndex"]?

Reply:
I missed `["address", "blockNumber", "transactionIndex"]`. I will fix it. Thanks.

Reply:
I added the index.

But I realized that I also missed changing the existing indexes for the `AddressLogs` and `AssetTypeLogs`. 

I will change them and ask for a review again.

Reply:
I changed the indexes of the `AddressLogs` and `AssetTypeLogs` tables.